frontend/src/views/StatblockEditorView.vue

```diff
@@ -686,23 +744,164 @@ export default defineComponent({
 				this.showSlides(moveToSlide);
 			}
 		},
-		changeCR(isIncrease: boolean): void {
-			let cr = this.data.description.cr;
-
-			if (cr == 0 && isIncrease) cr = 0.125;
-			else if (cr == 0.125 && isIncrease) cr = 0.25;
-			else if (cr == 0.25 && isIncrease) cr = 0.5;
-			else if (cr == 0.5 && isIncrease) cr = 1;
-			else if (cr == 0.125 && !isIncrease) cr = 0;
-			else if (cr == 0.25 && !isIncrease) cr = 0.125;
-			else if (cr == 0.5 && !isIncrease) cr = 0.25;
-			else if (cr == 1 && !isIncrease) cr = 0.5;
-			else {
-				if (isIncrease) cr = Math.min(30, cr + 1);
-				else cr = Math.max(0, cr - 1);
+		openCRCalculator(){
+			[this.crCalc.attackBonus, this.crCalc.dpr, this.crCalc.dc] = this.attackStats()
+
+			// Effective AC
+			this.crCalc.ac = this.calculateEffectiveAC()
+			this.crCalc.calculatedAC = this.crCalc.ac
+			
+			// Effective HP
+			this.crCalc.hp = this.calculateEffectiveHP()
+			this.crCalc.calculatedHP = this.crCalc.hp
+
+			this.crCalc.attackBonusCalc = this.crCalc.useDC == true ? this.crCalc.dc : this.crCalc.attackBonus
+			this.crCalc.expectedCR = this.data.description.cr
+			this.showCRModal = true
+		},
+		calculateEffectiveHP(): number{
+			let hp: number = this.data.defenses.hp.override ? this.data.defenses.hp.override : this.hpCalc()
+
+			// Adjust for vulnerabilities
+			if (this.data.defenses.vulnerabilities.length >0) hp = Math.floor(hp * this.crCalc.vulnerabilityModifier)
+
+			// Expected CR Multiplier			
+			hp = Math.floor(hp * this.expectedCRMultiplier())
+
+			return hp
+		},
+		expectedCRMultiplier(): number{
+			let multiplier = 1
+
+			if (this.data.defenses.immunities.length > 0){
```

Comment:
should immunities and resistances be mutually exclusive?

Reply:
Yeah. In the original calculator the immunities and resistances are a single dropdown #resistances where the value is either resistance or immunities